helm/alfresco-content-services/charts/activemq/templates/deployment-activemq.yaml

```diff
@@ -64,25 +68,11 @@ spec:
         resources:
 {{ toYaml .Values.resources | indent 12 }}
         volumeMounts:
-        - name: data
-          mountPath: {{ .Values.persistence.mountPath }}
-          subPath: {{ .Values.persistence.subPath }}
-      {{- if .Values.persistence.chownWithDynamicProvisioning }}
-      initContainers:
-        - name: init-fs
-          image: "{{ .Values.initContainer.image.repository }}:{{ .Values.initContainer.image.tag }}"
-          imagePullPolicy: {{ .Values.initContainer.image.pullPolicy }}
-          resources:
-{{ toYaml .Values.initContainer.resources | indent 14 }}
-          # command to allow repository to write to EFS volume.
-          command: ["sh", "-c", "chown -R 33031:1000 {{ .Values.persistence.mountPath }}"]
-          volumeMounts:
-            - name: data
-              mountPath: {{ .Values.persistence.mountPath }}
-              subPath: {{ .Values.persistence.subPath }}
-      {{- end }}
+        {{- if .Values.persistence.enabled }}
+          - name: data
+            mountPath: {{ .Values.persistence.data.mountPath }}
+            subPath: {{ .Values.persistence.data.subPath }}
       volumes:
-      - name: data
-        persistentVolumeClaim:
-          claimName: {{ .Values.persistence.existingClaim }}
+      {{- include "data_volume" .Values | nindent 8 }}
+      {{- end }}
```

Comment:
please avoiding spaning the same `if` across  several levels oy YAML. It's safer to have one under `volumeMounts` and another one under `volumes`